src/core/index.js

```diff
@@ -46,74 +51,50 @@ export default function SwaggerUI(opts) {
     buildTimestamp: BUILD_TIME,
   }
 
-  const domNode = opts.domNode
-  delete opts.domNode
+  const queryOptions = optionsFromQuery()(options)
+  let mergedOptions = mergeOptions({}, defaultOptions, options, queryOptions)
+  const storeOptions = storeOptionsFactorization(mergedOptions)
+  const InlinePlugin = inlinePluginOptionsFactorization(mergedOptions)
 
-  const { queryConfig, combinedConfig, storeConfig } = getConfigs(opts)
-
-  const inlinePlugin = ()=> {
-    return {
-      fn: combinedConfig.fn,
-      components: combinedConfig.components,
-      state: combinedConfig.state,
-    }
-  }
-
-  const store = new System(storeConfig)
-  store.register([combinedConfig.plugins, inlinePlugin])
 
+  const store = new System(storeOptions)
+  store.register([mergedOptions.plugins, InlinePlugin])
   const system = store.getSystem()
 
-  const downloadSpec = (fetchedConfig) => {
-    const localConfig = system.specSelectors.getLocalConfig ? system.specSelectors.getLocalConfig() : {}
-    const mergedConfig = deepExtend({}, localConfig, combinedConfig, fetchedConfig || {}, queryConfig)
-
-    // deep extend mangles domNode, we need to set it manually
-    if(domNode) {
-      mergedConfig.domNode = domNode
-    }
-
-    store.setConfigs(mergedConfig)
-    system.configsActions.loaded()
-
-    if (fetchedConfig !== null) {
-      if (!queryConfig.url && typeof mergedConfig.spec === "object" && Object.keys(mergedConfig.spec).length) {
-        system.specActions.updateUrl("")
-        system.specActions.updateLoadingStatus("success")
-        system.specActions.updateSpec(JSON.stringify(mergedConfig.spec))
-      } else if (system.specActions.download && mergedConfig.url && !mergedConfig.urls) {
-        system.specActions.updateUrl(mergedConfig.url)
-        system.specActions.download(mergedConfig.url)
+  const configURL = queryOptions.config ?? mergedOptions.configUrl
+  const yamlFileOptions = optionsFromYAMLFile({ system })(mergedOptions)
+
+  optionsFromURL({ url: configURL, system })(mergedOptions)
+    .then((urlOptions) => {
+      const urlOptionsFailedToFetch = urlOptions === null
+
+      mergedOptions = mergeOptions({}, yamlFileOptions, mergedOptions, urlOptions, queryOptions)
```

Comment:
Adding this comment here as well with updates:

Doesn't this order of merging mean that we will always overwrite values in `yamlFileOptions` with the `defaultOptions` values which are merged into `mergedOptions`?

We also can't specify `configUrl` inside the YAML file as it's fetched from the `mergedOptions` before.

If that's the case and we need to change this, we can do it in my PR as well.

Reply:
> Doesn't this order of merging mean that we will always overwrite values in yamlFileOptions with the defaultOptions values which are merged into mergedOptions?

Not with the `defaultOptions`, but with options created from a merge of `default, user and query` options. These options gets higher precedence then `yamlFileOptions` and then the `urlOptions` and `queryOptions` kicks in to apply their highest precedence.

I would agree it looks strange, but overall makes sense to me

Every options source has it's priority and given following priority order, the above code is correct:

(highest options source is more important)
1. query options
2. url options
3. user options
4. yaml file options

Reply:
> If that's the case and we need to change this, we can do it in my PR as well.

We shouldn't do any functionality changes in consolidation PRs. If we want to change or fix something, we need to create separate issues, will full description and motivations and follow with separate PRs. This consolidation should be 100% backward compatible.

Reply:
Yes, `user` and `query` options should take precedence, but what I meant is a case like:
1. we define some options in YAML file
2. we don't define anything for query and user options so `mergedOptions` only has values from `defaultOptions`

so wouldn't doing the deepExtend mean that `defaultOptions` would overwrite `yamlFileOptions`?

Reply:
We'll rename yaml-file options into `system` and we'll create a separate issue for order of precendence.

Reply:
We'll create a separate issue for order of precedence.

```js
mergedOptions = mergeOptions({}, defaultOptions, systemOptions, userOptions, urlOptions, queryOptions)
```